#161250654 User should be able to save articles as draft

kampkelly opened the pull request:
#### What does this PR do?
This pull request adds the saving of articles as draft

#### Description of Task to be completed?
- Write the code to save article as draft

#### How should this be manually tested?
Using postman and as a signed in user, make requests to the following api endpoints.

To save an article as draft: /api/v1/article -> POST request

#### Any background context you want to provide?
None
#### What are the relevant pivotal tracker stories?
#161250654
#### Questions:```
None

kampkelly committed:
feature(draft-article): save draft article

This commit enables saving of an article as draft. When an article is created, it is saved as draft by default.

Finishes #161250654

kampkelly committed:
feature(draft-article): add test

This commit adds a test

s-akinrele commented on server/controllers/articleController.js:
```diff
@@ -107,10 +107,15 @@ export default class ArticleController {
   static getUserArticles(req, res) {
     const page = computeOffset(req);
-
+    let whereParams;
```
update this variable name

kampkelly committed:
feature(draft-article): validate status

This commit adds validations and tests for the status column while saving an article.

kampkelly committed:
Merge branch 'staging' of github.com:andela/phlox-ah into feature/161250654/save-draft-articles

# Conflicts:
#	server/controllers/articleController.js
#	server/routes/api/article.js
#	server/tests/categories.test.js

kampkelly committed:
feature(draft-article): refactor code

This commit refactors code.

Finishes #161250654

kampkelly committed:
feature(draft-article): add category to seed

This commit adds category to the seed.
